_docs/geyser/extensions.md

```diff
@@ -0,0 +1,116 @@
+
+### What can Geyser Extensions do?
+Extensions can fully utilize the Geyser API to add additional functionality to Geyser. See [Geyser API docs](/geyser/geyser-api/) for info on what is currently possible. To name a few examples:
+- Register custom items
+- Hide commands from being suggested
+- Change the MOTD
+- Register your own commands
+- Listen to events, e.g. the emote event.
+
+The underlying Geyser API is steadily expanding, creating more and more opportunities.
+
+### Which Geyser Extensions exist?
+At the moment, not a lot. This is due to Extensions being quite new, and that most stuff can already be done with "normal" plugins/mods.
+There is an [official list](https://github.com/GeyserMC/GeyserExtensionList) of available extensions you can check out.
+
+### Installing Extensions
+To install an extension, simply put the extension .jar file into Geyser's 'extensions' folder. Then, restart Geyser (or the server Geyser runs on).
+
+### Creating Geyser Extensions
+The easiest way to create an extension would be using [this official template](https://github.com/GeyserMC/GeyserExampleExtension/). Simply create a new repository from the template, customize the 'extension.yml' & 'settings.gradle' files, and get started making the extension.
+
+Geyser recognizes extensions when they have a file called 'extension.yml' in the jars 'resources' folder.
+
+extension.yml:
+```yml
+id: exampleid
+name: ExampleExtension
+main: org.geyser.extension.exampleid.ExampleExtension
+api: 2.1.1
+version: 1.0.0
+authors: [ExampleAuthor]
+```
+
+Explanations for the individual fields:
+- id: The id of the extension. Every extension needs to have their unique id - all lowercase letters. If you e.g. register a command for your extension, it will have the id as a prefix: e.g. '/exampleid command'.
+- name: The name of the extension.
+- main: The main class of your extension.
+- api: The Geyser api version your extension targets.
+- authors: The author(s) of the extension. To add more than one entry, separate entries with a comma.
+
+The main class, the entrypoint for the extension, needs to [implement the 'Extension' interface provided by Geyser](https://github.com/GeyserMC/GeyserExampleExtension/blob/47614575a69bddecb241676215f3c9f9113db304/src/main/java/org/geyser/extension/exampleid/ExampleExtension.java#L10). 
+That way, Geyser recognizes the extension, and gives you access to important methods - such as 'logger()', to get your extensions logger. <br>
+To see all the methods provided by that interface, see [here](https://github.com/GeyserMC/Geyser/blob/master/api/src/main/java/org/geysermc/geyser/api/extension/Extension.java).
+
+If you wish to register custom items, global resource packs (or soon, custom blocks and entities), you would need to subscribe to the event using the @Subscribe annotation, 
```

Comment:
This should be moved further down. Would also note the distinction between other platforms (i.e. Spigot) that use `onEnable` and `onDisable` and the fact that extensions uses events. I'd also include a bit about the other lifecycle events, like preinitialize and shutdown.